**Helper functions**: A few functions that are useful whilst working with simplex tableaus.

In [11]:
# Add the column and row numbers to the matrix for easier indexing.
# It preserves the original naming.
colrownames <- function(M) {
    cn=colnames(M)
    cn=paste(cn, ' (', c(1:length(M[1,])), ')', sep='')
    colnames(M)=cn
    rn=rownames(M)
    rn=paste(rn, ' (', c(1:length(M[,1])), ')', sep='')
    rownames(M)=rn
    M
}

# Returns the pivot element for a given column.
find.pivot <- function(M, col) {
    rhs.index=length(M[,1])
    row=M[,rhs.index]/M[,col]
    min(row[row>0 & !is.na(row) & row!=Inf])
}

# LP model

**Beslissingsvariabelen**

* $x_1$ : Oliesoort 1
* $x_2$ : Oliesoort 2
* $x_3$ : Oliesoort 3

**LP probleem**

$
\max 12x_1 + 15x_2 + 20x_3 \\
x_1 + 4x_2 + 8x_3 \leq 1000 \\ 
0.5x_1+2x_2+0.25x_3 \leq 1500 \\
7x_1+6x_2+x_3\leq 1400 \\
3x_1+5x_2+4x_3 \leq 2000
$

**Canonieke vorm**

$
A+A_5=0 \\
d-12x_1-15x_2-20x_3=0 \\
$

# Solving

In [3]:
A=c(1,0,0,0,0,0,0,0)
d=c(0,1,0,0,0,0,0,0)
x1=c(-1,-12,1,.5,7,3,1,0)
x2=c(0,-15,4,2,6,5,0,0)
x3=c(0,-20,8,.25,1,4,0,1)
s1=c(0,0,1,0,0,0,0,0)
s2=c(0,0,0,1,0,0,0,0)
s3=c(0,0,0,0,1,0,0,0)
s4=c(0,0,0,0,0,1,0,0)
a5=c(0,0,0,0,0,0,1,0)
s5=c(1,0,0,0,0,0,-1,0)
s6=c(0,0,0,0,0,0,0,1)
RHS=c(-50,0,1000,1500,1400,2000,50,10)
M=cbind(A,d,x1,x2,x3,s1,s2,s3,s4,a5,s5,s6,RHS)
rownames(M)=c('A','d','s1','s2','s3','s4','s5','s6')
M=colrownames(M)

# Iteratie 2
M[1,]=M[1,]+M[7,]
M[2,]=M[2,]+12*M[7,]
M[3,]=M[3,]-M[7,]
M[4,]=M[4,]-1/2*M[7,]
M[5,]=M[5,]-7*M[7,]
M[6,]=M[6,]-3*M[7,]

# Iteratie 3
M[2,]=M[2,]+20*M[8,]
M[3,]=M[3,]-8*M[8,]
M[4,]=M[4,]-1/4*M[8,]
M[5,]=M[5,]-M[8,]
M[6,]=M[6,]-4*M[8,]

# Iteratie 4
M[5,]=M[5,]/6
M[2,]=M[2,]+15*M[5,]
M[3,]=M[3,]-4*M[5,]
M[4,]=M[4,]-2*M[5,]
M[6,]=M[6,]-5*M[5,]
M

A (1) d (2) x1 (3) x2 (4) x3 (5) s1 (6) s2 (7) s3 (8)     s4 (9)
A (1)  1     0     0      0      0      0      0       0.0000000 0     
d (2)  0     1     0      0      0      0      0       2.5000000 0     
s1 (3) 0     0     0      0      0      1      0      -0.6666667 0     
s2 (4) 0     0     0      0      0      0      1      -0.3333333 0     
s3 (5) 0     0     0      1      0      0      0       0.1666667 0     
s4 (6) 0     0     0      0      0      0      0      -0.8333333 1     
s5 (7) 0     0     1      0      0      0      0       0.0000000 0     
s6 (8) 0     0     0      0      1      0      0       0.0000000 0     
       a5 (10)   s5 (11)   s6 (12)     RHS (13) 
A (1)   1.000000  0.000000  0.00000000    0.0000
d (2)  -5.500000  5.500000 17.50000000 3400.0000
s1 (3)  3.666667 -3.666667 -7.33333333  176.6667
s2 (4)  1.833333 -1.833333  0.08333333 1125.8333
s3 (5) -1.166667  1.166667 -0.16666667  173.3333
s4 (6)  2.833333 -2.833333 -3.16666667  943.3333
s5 (7)  1.000000 -1.000000  0.00000000   50.0000
s6 (8)  0.000000  0.000000  1.00000000   10.0000

De oplossing is $(x_1,x_2,x_3)=(50, 173.33, 10)$.

## 42.a

_Als volgende week vanwege een strengere milieuwetgeving van oliesoort 3 nog maar
hoogstens 4 ton gemaakt mag worden, wat zou dan de optimale oplossing en maximale
winst zijn?_

De vraagstelling heeft betrekking tot de beperking: $x_3 \leq 10$. Dit wordt namelijk $x_3 \leq 4$. In de canonieke vorm is dit $x_3 + s_3 = 10$. Hier moet $k$ van af, dus $s_3^*=s_3-k$.

In [6]:
M[,13]-6*M[,12]

A (1)     d (2)    s1 (3)    s2 (4)    s3 (5)    s4 (6)    s5 (7)    s6 (8) 
   0.0000 3295.0000  220.6667 1125.3333  174.3333  962.3333   50.0000    4.0000

De nieuwe oplossing is $(x_1,x_2,x_3)=(50, 174.33, 4)$ met een maximale winst van $€3295$.

## 42.b

_Stel, de fabriek wil volgende week minstens 170 ton van oliesoort 1 maken. Wat
zou dan de optimale oplossing en maximale winst zijn?_

In [9]:
M[,13]-120*M[,11]

A (1)      d (2)     s1 (3)     s2 (4)     s3 (5)     s4 (6)     s5 (7) 
   0.00000 2740.00000  616.66667 1345.83333   33.33333 1283.33333  170.00000 
    s6 (8) 
  10.00000

## 42.c

_Stel, de fabriek wil volgende week minstens 200 ton van oliesoort 1 maken. Wat
zou dan de optimale oplossing en maximale winst zijn?_

In [10]:
M[,13]-150*M[,11]

A (1)       d (2)      s1 (3)      s2 (4)      s3 (5)      s4 (6) 
   0.000000 2575.000000  726.666667 1400.833333   -1.666667 1368.333333 
     s5 (7)      s6 (8) 
 200.000000   10.000000

Dit ligt buiten het interval, maar wat betekend dat?